flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverStrategyLoader.java

```diff
@@ -44,13 +44,18 @@
 	/** Config name for the {@link RestartPipelinedRegionStrategy}. */
 	public static final String LEGACY_PIPELINED_REGION_RESTART_STRATEGY_NAME = "region-legacy";
 
+	/** Config name for the {@link NoOpFailoverStrategy} */
+	public static final String NO_OP_FAILOVER_STRATEGY = "noop";
+
 	// ------------------------------------------------------------------------
 
 	/**
 	 * Loads a FailoverStrategy Factory from the given configuration.
 	 */
 	public static FailoverStrategy.Factory loadFailoverStrategy(Configuration config, @Nullable Logger logger) {
-		final String strategyParam = config.getString(JobManagerOptions.EXECUTION_FAILOVER_STRATEGY);
+		final String strategyParam = config.getString(JobManagerOptions.SCHEDULER).equals("ng") ?
+			NO_OP_FAILOVER_STRATEGY :
```

Comment:
How about a `ThrowingFailoverStrategy` here, just like `ThrowingRestartStrategy`? We do not expect the strategy to be used after the creation. `ThrowingFailoverStrategy` can help to found out potential issues.

We should use `isLegacyScheduling()` check to surround the `failoverStrategy#notifyNewVertices` invocation in `ExecutionGraph#attachJobGraph` as well.

Reply:
I would leave it as is for now. It was done because `buildGraph` already invokes `notifyNewVertices`. At the point of `buildGraph`, we still do not know if legacy scheduling is enabled or disabled.

Reply:
I see.